tests/kernel/spinlock/src/spinlock_error_case.c

```diff
@@ -14,6 +14,12 @@ static struct k_spinlock lock;
 static struct k_spinlock mylock;
 static k_spinlock_key_t key;
 
+/* Needs to be allocated here for coherence */
```

Comment:
This isn't sufficient.  The coherence issue is specific/unique to some systems and we need to state that.  For example, I'm guessing this works perfectly fine as is on a cortex-m0/m3/m4 system.

Reply:
I'm not sure I understand the issue here exactly. The comment should be added to all spin locks that they need to be global because KERNEL_COHERENCE dictates that and spin locks assert on it? Just this one particular spin lock because I made a mistake and CI doesn't run on our hardware? What are you looking for here, please be specific because to me the comment is sort of meaningless. It's a known, and assert on, attribute of this platform.

Reply:
I'm looking for a comment that states something probably about xtensa memory regions and which ones are coherent.  On some ARM64 system that is fully coherent the old code would work.  So I'm saying the comment should explain why this needs can't just be a local variable.

Reply:
Done